src/__tests__/components/ShoppingCart/ShoppingCartContainer.test.tsx

```diff
@@ -78,4 +78,36 @@ describe("ShoppingCart component", () => {
       `height:auto ; overflow-y:scroll; white-space:normal; text-overflow:clip `
     );
   });
+
+  it("Should render expired message", () => {
+    renderWithMockedProvider(<ShoppingCartContainer shoppingCart={mockCartData} />);
+    const expiredMessage = screen.getByText(
+      "Sorry, some product’s quotation has expired. Please edit your cart and try again. We’re apologize for any inconvenience caused."
+    );
+    expect(expiredMessage).toBeVisible();
+  });
+
+  it("Should render expired icon and expired quotation message", () => {
+    renderWithMockedProvider(<ShoppingCartContainer shoppingCart={mockCartData} />);
+    const listItems = screen.queryAllByRole("dataRow");
+    mockCartData.forEach((item, index) => {
+      if (item.isExpired) {
```

Comment:
it's better to avoid `if else` in a test case, you can split it into two cases.

Reply:
Thanks Molly. I've updated the test:)